config/initializers/hostname.rb

```diff
@@ -1,16 +1,17 @@
-# Compute hostname and port from the hostname in the Settings
+# Compute hostname and port from the hostname in the Settings.
+# If no hostname is specified, read the hostname from the OS.
 module Hostname
   def self.fqdn
-    hostname = Settings.hostname.split(':').first
-    if hostname == 'localhost' || hostname.include?('.')
+    hostname = Settings.hostname.split(':').first if Settings.hostname
+    if hostname == 'localhost' || (hostname && hostname.include?('.'))
       hostname
     else
-      Addrinfo.tcp(tmp, 0).getnameinfo.first
+      Addrinfo.tcp(Socket.gethostname, 0).getnameinfo.first
```

Comment:
Shouldn't this fail with a more useful error message than `SocketError: getaddrinfo: nodename nor servname provided, or not known`?